Packs/RubrikPolaris/Integrations/RubrikPolaris/RubrikPolaris_test.py

```diff
@@ -2350,3 +2350,311 @@ def test_gps_vm_recover_files_command_success(client, requests_mock, empty_respo
         assert response.raw_response == gps_vm_recover_files_response.get('raw_response')
         assert response.outputs == gps_vm_recover_files_response.get('outputs')
         assert response.readable_output == gps_vm_recover_files_hr
+
+
+@pytest.mark.parametrize("empty_response, limit, page_number",
+                         [(True, 1, 1), (False, 1, 1), (False, 1, 2), (False, 2, 1)])
+def test_rubrik_sonar_user_access_list_command_success(client, requests_mock, empty_response, limit, page_number):
+    """
+    Test case scenario for rubrik_sonar_user_access_list_command with valid case.
+
+    When:
+        - Calling rubrik_sonar_user_access_list_command.
+    Then:
+        - Verifies mock response with actual response.
+    """
+    from RubrikPolaris import rubrik_sonar_user_access_list_command
+
+    # Load test data
+    response_data = util_load_json(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                                "test_data/sonar_user_access_list_response.json"))
+    with open(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                           f"test_data/sonar_user_access_list_response_hr_{limit}.md"), 'r') as f:
+        hr_data = f.read()
+
+    args = {"sort_order": "ASC", "limit": limit, "page_number": page_number, "include_whitelisted_results": True,
+            "user_email": "demo"}
+    if empty_response:
+        requests_mock.post(BASE_URL_GRAPHQL, [{"json": response_data.get('empty_response')}])
+        response = rubrik_sonar_user_access_list_command(client, args=args)
+        assert response.readable_output == MESSAGES['NO_RECORDS_FOUND'].format('user accesses')
+    else:
+        requests_mock.post(BASE_URL_GRAPHQL, [{"json": response_data.get('raw_response')}])
+        response = rubrik_sonar_user_access_list_command(client, args=args)
+        outputs = response_data.get('outputs')
+        page_token = response_data.get(f'page_token_{limit}')
+        if limit == 1:
+            outputs = [outputs[page_number - 1]]
+            if page_number == 2:
+                hr_data = hr_data.replace("1-1 out of 2", "2-2 out of 2")
+                hr_data = hr_data.replace("demo", "demo2")
+                hr_data = hr_data.replace("Demo", "Demo2")
+                hr_data = hr_data.replace("0001", "0002")
+                hr_data += f"\n{MESSAGES['NEXT_PAGE_TOKEN'].format('cursor_2')}"
+                page_token.update({"has_next_upn_page": False})
+        assert response.raw_response == response_data.get('raw_response')
+        assert response.outputs.get(f'{OUTPUT_PREFIX["USER_ACCESS"]}(val.principalId == obj.principalId)') == \
+               remove_empty_elements(outputs)
+        assert response.outputs.get(f'{OUTPUT_PREFIX["PAGE_TOKEN_USER_ACCESS"]}(val.name == obj.name)') == \
+               remove_empty_elements(page_token)
+        assert response.readable_output == hr_data
```

Comment:
We prefer not to share to cases in the tests results.. It will be better if we add a different case for empty results and add more options to the mark.parametrize to assert with rather than 'build' the expected result inside the test

Reply:
Done